libportal/remote.c

```diff
@@ -41,6 +41,139 @@ typedef struct {
   guint cancelled_id;
 } CreateCall;
 
+struct _XdpScreenCastSessionClass
+{
+  XdpSessionClass parent_class;
+};
+
+typedef struct {
+  GVariant *streams;
+  XdpPersistMode persist_mode;
+  char *restore_token;
+} XdpScreenCastSessionPrivate;
+
+G_DEFINE_TYPE_WITH_PRIVATE (XdpScreenCastSession, xdp_screencast_session, XDP_TYPE_SESSION)
```

Comment:
nit: it's `screen_cast` and `remote_desktop` elsewhere, not `screencast` and `remotedesktop`. Might cause confusion if we mix and match.